src/bindings/fluid_cmd.c

```diff
@@ -2133,27 +2135,25 @@ int fluid_handle_channelsmode (void* data, int ac, char** av,
 		{
 			if(bci.mode &  FLUID_CHANNEL_ENABLED)
 			{	/* This channel is enabled */
-				char * basic_channel="basic channel"; /* field basic channel */
 				char * bc_msg; /* field basic channel */
-				char * poly_msg ="poly"; /* field mode */
-				char * mono_msg ="mono"; /* field mode */
 				char * p_mode; /* field mode */
 				char * blank="--";
-				char nbr[10]; /* field Nbr */
 				char *p_nbr; /* field Nbr */
 				int mode = bci.mode &  FLUID_CHANNEL_MODE_MASK;
 				if (bci.mode &  FLUID_CHANNEL_BASIC)
 				{	/* This channel is a basic channel */
-					bc_msg = basic_channel;
-					sprintf(nbr,"nbr:%3d",bci.val);
+					char nbr[10]; /* field Nbr */
+					bc_msg = "basic channel";
+					nbr[sizeof(nbr)-1] = 0;
+					FLUID_SNPRINTF(nbr,sizeof(nbr)-1,"nbr:%3d",bci.val);
```

Comment:
Manual null termination is not necessary for c99 snprintf, you can use the full buffer.

Reply:
Sorry, when using  this example:
```
 char nbr[4]; /* field Nbr */
 nbr[sizeof(nbr)-1] = 0;					
 snprintf(nbr,sizeof(nbr),"%d",9999); 
 if(nbr[sizeof(nbr)-1] != 0) 
 {        
  printf "not null termined"; 
 }
```
The message "not null termined" is displayed.  What could we do ?

Reply:
Are you compiling this with VC? It seems like VC returns an error code if the buffer is too small to fit the formatted string + null byte: https://github.com/magnumripper/JohnTheRipper/issues/851
Maybe we need a similar (if  not the same) shim for old VC versions?

Reply:
Seems like @jjceresa trapped into microsofts non standard, non compliant, shitty extension `_snprintf()`. That's why I did this yesterday to master: 272b8073833b1b1867caf71759924617592e1b2f

...perhaps it's work cherry-picking it to `1.1.x`.

Reply:
>  That's why I did this yesterday to master: 272b807

Ah, that's much better. I must say I really like having glib around :-)

Reply:
> I must say I really like having glib around

That's so true. It is still very sad that microsoft needed until VS2015 for a c99 compliant library implementation.